clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -110,6 +110,26 @@ void SYCL::constructLLVMForeachCommand(Compilation &C, const JobAction &JA,
         C.getArgs().MakeArgString("--out-dir=" + OutputDirName));
   }
 
+  // If fsycl-dump-device-code is passed, put the PTX files
+  // into the path provided in fsycl-dump-device-code.
+  if (T->getToolChain().getTriple().isNVPTX() &&
+      C.getDriver().isDumpDeviceCodeEnabled() && Ext.equals("s")) {
+    SmallString<128> OutputDir;
+
+    Arg *DumpDeviceCodeArg =
+        C.getArgs().getLastArg(options::OPT_fsycl_dump_device_code_EQ);
+
+    OutputDir = (DumpDeviceCodeArg ? DumpDeviceCodeArg->getValue() : "");
+
+    // If the output directory path is empty, put the PTX files in the
+    // current directory.
+    if (OutputDir.empty())
+      llvm::sys::path::native(OutputDir = "./");
+    else
+      OutputDir.append(llvm::sys::path::get_separator());
```

Comment:
Can you explain why the ending path separator is needed again?

Reply:
I followed the same template that was in place for `--save-temps` implementation, where they appended a path separator if the output directory wasn't empty.
I removed the append, and the PTX files were saved in the output directory input by the user on Linux. 
I don't have the CUDA toolkit installation on Windows, so not sure if the actual compilation works there too. Just invoking -### seems to work on both Linux and Windows.
Example: 
```
-fsycl-dump-device-code=/x/y/z     ==> --out-dir= /x/y/z
-fsycl-dump-device-code=/x/y/z/     ==> --out-dir= /x/y/z/
```

Reply:
Thanks.  The behavior is not hurting anything - just was odd that we added it.